tidy3d/components/material/tcad/charge.py

```diff
@@ -309,10 +312,66 @@ class SemiconductorMedium(AbstractChargeMedium):
     )
 
     N_d: Union[pd.NonNegativeFloat, SpatialDataArray, tuple[DopingBoxType, ...]] = pd.Field(
-        0,
+        (ConstantDoping(concentration=0, center=(0, 0, 0), size=(inf, inf, inf)),),
         title="Doping: Donor concentration",
         description="Concentration of donor impurities, which create mobile electrons, resulting in n-type material. "
         "Can be specified as a single float for uniform doping, a :class:`SpatialDataArray` for a custom profile, "
         "or a tuple of geometric shapes to define specific doped regions.",
         units=PERCMCUBE,
     )
+
+    # DEPRECATION VALIDATORS
+    @pd.validator("N_c", always=True)
+    def check_nc_uses_model(cls, val, values):
+        """Issue deprecation warning if float is provided"""
+        if isinstance(val, (float, int)):
+            log.warning(
+                "Passing a float to 'N_c' is deprecated and will be removed in future versions. "
+                "Please use 'ConstantEffectiveDOS' instead."
+            )
+            return ConstantEffectiveDOS(N=val)
+        return val
+
+    @pd.validator("N_v", always=True)
+    def check_nv_uses_model(cls, val, values):
+        """Issue deprecation warning if float is provided"""
+        if isinstance(val, (float, int)):
+            log.warning(
+                "Passing a float to 'N_v' is deprecated and will be removed in future versions. "
+                "Please use 'ConstantEffectiveDOS' instead."
+            )
+            return ConstantEffectiveDOS(N=val)
+        return val
+
+    @pd.validator("E_g", always=True)
+    def check_eg_uses_model(cls, val, values):
+        """Issue deprecation warning if float is provided"""
+        if isinstance(val, (float, int)):
+            log.warning(
+                "Passing a float to 'E_g' is deprecated and will be removed in future versions. "
+                "Please use 'ConstantEnergyBandGap' instead."
+            )
+            return ConstantEnergyBandGap(eg=val)
+        return val
+
+    @pd.validator("N_d", always=True)
+    def check_nd_uses_model(cls, val, values):
+        """Issue deprecation warning if float is provided"""
+        if isinstance(val, (float, int)):
+            log.warning(
+                "Passing a float to 'N_d' is deprecated and will be removed in future versions. "
+                "Please use 'SpatialDataArray' or a list of 'DopingBoxType' instead."
```

Comment:
I feel like here we should just say use `ConstantDoping` instead because if the user is providing a float, that's the equivalent way. Or at least something like "use the equivalent  'ConstantDoping', or other 'DopingBoxType' classes or a 'SpatialDataArray' for more advanced doping definitions."